### ISSUE TITLE
HKG - `#asWord("")` is unexpected during the proof

### ISSUE THREAD

**Issue by yzhang90** [kframework/kore#1525]
Following the issue #1520, we encountered the warnings:
```
kore-exec: [2020-02-06 20:57:54.711656538] Warning (WarnBottomHook): Evaluating 'BYTES.get' in pattern:
    /* Fn Spa */
    Lbl'UndsLSqBUndsRSqBUnds'BYTES-HOOKED'Unds'Int'Unds'Bytes'Unds'Int{}(
        /* Fl Fn D Sfa Cl */ \dv{SortBytes{}}(""),
        /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
    )
has resulted in \bottom.
```
This happens when the prover tries to apply the rule
```
    syntax PushOp ::= PUSH ( Int )
 // ------------------------------
    rule <k> PUSH(N) => #asWord(PGM [ PCOUNT +Int 1 .. N ]) ~> #push ... </k>
         <pc> PCOUNT </pc>
         <program> PGM </program>
at step 99.
```
We suspect that the prover tries to apply the following lemma
```
    rule #asWord( BUF => #drop(1, BUF) ) requires BUF [ 0 ] ==Int 0  [simplification]
```
on the term `asWord("")`.

However, the opcode is PUSH1 and the result buffer should be a bytearray of length 1 instead of "".

**Comment by yzhang90** [comment]
After @virgil-serbanuta show me the function call stack log, I made the following guess of how `#asWord("")` is encountered during the execution.
First, `#asWord(0)` is encountered during the execution (which is expected), and then the following simplification rule is applied before the definition rule.
```
rule #asWord( BUF => #drop(1, BUF) ) requires BUF [ 0 ] ==Int 0  [simplification]
```
The result of the simplication is `#asWord("")`, and then the same simplification rule is tried again and leads to the warnings.

If my guess is correct, the problem should be fixed by #1524

**Comment by ttuegel** [comment]
@yzhang90 Was this fixed by #1524.

**Comment by yzhang90** [comment]
yes. I closed the issue.
